src/create-text-tracks-if-necessary.js

```diff
@@ -16,14 +16,34 @@ const createTextTracksIfNecessary = function(sourceBuffer, mediaSource, segment)
 
   // create an in-band caption track if one is present in the segment
   if (segment.captions &&
-      segment.captions.length &&
-      !sourceBuffer.inbandTextTrack_) {
-    removeExistingTrack(player, 'captions', 'cc1');
-    sourceBuffer.inbandTextTrack_ = player.addRemoteTextTrack({
-      kind: 'captions',
-      label: 'cc1'
-    }, false).track;
-    player.tech_.trigger({type: 'usage', name: 'hls-608'});
+      segment.captions.length) {
+    if (!sourceBuffer.inbandTextTracks_) {
+      sourceBuffer.inbandTextTracks_ = {};
+    }
+
+    for (let trackId in segment.captionTracks) {
+      if (!sourceBuffer.inbandTextTracks_[trackId]) {
+        player.tech_.trigger({type: 'usage', name: 'hls-608'});
+        let track = player.textTracks().getTrackById(trackId);
+
+        if (track) {
+          // Resuse an existing track with a CC# id because this was
+          // very likely created by videojs-contrib-hls from information
+          // in the m3u8 for us to use
+          sourceBuffer.inbandTextTracks_[trackId] = track;
+        } else {
+          // Otherwise, create a track with the default `CC#` label and
+          // without a language
+          removeExistingTrack(player, 'captions', trackId);
```

Comment:
Right, cool. Yeah in that case we should never have a situation where we have a `CC1` text track that has the wrong language/label.

Removing it caused [this test](https://github.com/squarebracket/videojs-contrib-media-sources/blob/d07b3a215ce5785b59ed4cf938b901ecb69bafd9/test/html.test.js#L1233) to fail, which makes sense. Think it's safe to just adapt the test, will deleting the `removeExistingTrack` have other possible side effects?